x-pack/plugins/fleet/server/services/epm/elasticsearch/template/template.ts

```diff
@@ -534,9 +534,15 @@ export function generateTemplateName(dataStream: RegistryDataStream): string {
 /**
  * Given a data stream name, return the indexTemplate name
  */
-function dataStreamNameToIndexTemplateName(dataStreamName: string): string {
-  const [type, dataset] = dataStreamName.split('-'); // ignore namespace at the end
-  return [type, dataset].join('-');
+async function getIndexTemplate(
+  esClient: ElasticsearchClient,
+  dataStreamName: string
+): Promise<string> {
+  const dataStream = await esClient.indices.getDataStream({
```

Comment:
Just double checking, does this getDataStream call work even if the stream has no data?

Reply:
good question, I'll check

Reply:
checked and if the data stream doesn't exist, it is not hitting this code path